src/Cart.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { deleteOrderProductsThunk, updateOrderProductThunk, se } from './store';
+import { setOrderProductsThunk, setOrdersThunk, setProductsThunk, setUsersThunk} from './store'
+
+
+
+class _Cart extends React.Component {
+  constructor(props) {
+    super();
+    this.deleteItem = this.deleteItem.bind(this);
+    this.updateItem = this.updateItem.bind(this);
+  }
+  // async componentDidMount() {
+  //   await this.props.getUsers()
+  //   await this.props.getProducts()
+  //   await this.props.getOrders()
+  //   await this.props.getOrderProdcuts()
+  // }
+  async deleteItem (id){
+    await this.props.deleteItem(id)
+  }
+  async updateItem (item) {
+    await this.props.updateItem(item)
+  }
+  render(){
+    const { orders , products, orderProducts, auth, match } = this.props;
+    // console.log('orders', orders)
+    const cart = orders.find(order => order.userId === match.params.id && order.status ==='cart');
+    if ( cart === undefined){
+      return('You have no cart at this time.')
+    }
+    // console.log('cart', cart)
+    // console.log('match', match)
+    // console.log('auth', auth)
+    const cartItems = orderProducts.filter(item => item.orderId === cart.id);
```

Comment:
This is the same as earlier, I think what we want is one backend route that returns something like:

```typescript
interface Cart {
  orderProducts: OrderProduct[]
}
```

Where this is the one and only cart that the user has, and it already has all of its relevant orderProducts in it.